### PR TITLE
Add catalog/vappTemplate/Media support and add option to define the a…

### PR DESCRIPTION
**Pull request by JoMas971**
Add catalog/vappTemplate/Media support and add option to define the Api/KC urls in env.

### CODE REVIEW

File: src/sdk/basic-configuration.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    } else {
+      return 'https://api.ilandcloud.com';
+    }
+  }
+
+  /**
+   * Get the default keycloak authorisation url.
+   * @returns {string}
+   */
+  static getKeycloakUrl(): string {
```
Comment (cfsnyder) [inline src/sdk/basic-configuration.ts]: Lets call it `getAuthorizationUrl()` and lets update the environment variable, etc to be generic as well

File: src/sdk/model/__tests__/vapp-template.test.unit.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+beforeAll(() => {
+  Iland.init(new IlandDirectGrantAuthProvider({
+    username: '',
+    password: '',
+    clientSecret: '',
+    clientId: ''
+  }));
+});
+
+function runMediaAssertionsAgainsMock(vappTemplate: VappTemplate) {
```
Comment (cfsnyder) [inline src/sdk/model/__tests__/vapp-template.test.unit.ts]: misnamed

File: src/sdk/basic-configuration.ts (hunk 1 above)
Comment (JoMas971) [reply src/sdk/basic-configuration.ts]: Ok let me change the name though.

File: src/sdk/model/__tests__/vapp-template.test.unit.ts (hunk 2 above)
Comment (JoMas971) [reply src/sdk/model/__tests__/vapp-template.test.unit.ts]: Nice catch thanks ! Will update this.

**Comment by JoMas971** [comment]
Should be good now. I've fixed your comments.

**Pull request merged**